### ISSUE TITLE
My service-worker.js doesn't exist when the mode equals "development"

### ISSUE THREAD

**Issue by sneko** [vuejs/vue-cli#2678]
### Version
3.0.0-rc.5

### Node and OS info
yarn

### Steps to reproduce
I have 2 environments `production` and `development` and for each one I build my Vue app with `yarn run build --mode development` or `yarn run build --mode production`.

I would like to test PWA in development environment but I noticed that the `service-worker.js` was not created in `development` mode. So my browser (Chrome) is not able to test that even if the content of the file seems minified in one the `app.js` aside chunk files (`1.js`, `2.js`, ...).

How do you manage that to test PWA please?

### What is expected?
PWA working even in development mode

### What is actually happening?
Not working

**Comment by LinusBorg** [comment]
> 

`yarn serve --mode production`

**Comment by sneko** [comment]
@LinusBorg I would like to deploy it on my cluster in an Nginx image to simulate production condition (except code is not minified and is viewable to debug).

Is there a workaround by building the project?

**Comment by LinusBorg** [comment]
Well you can disable the minification by reconfiguring webpack:
```javascript
// in vue.config.js
chainWebpack: config => {
  if (process.env.NO_MINIMIZE) {
    config.optimization.minimize(false)
    config.optimization.delete('minimizer')
  }
}
```
Then run it like this:
```
NO_MINIMIZE=true vue-cli-service build
```

**Comment by mc6415** [comment]
> That's designed in purpose so the service worker cache doesn't mess with your hot module reloading and stuff.
> 
> > How do you manage that to test PWA please?
> 
> The simplest way would be to run
> 
> `yarn serve --mode production`
> 
> Or just build the app and serve it though e.g. `http-server`.

Googling around this was the only thing I could find however using `yarn serve --mode production` is still yielding the replacement service-worker.js file, I was wondering if you might have any ideas why this was?

**Comment by OwenMelbz** [comment]
@mc6415 you found a solution? Been searching for hours how to get this to work, there doesn't seem to be anywhere that says what exactly the pwa plugin does, or where it stops and how to continue :(

**Comment by Phen-Ro** [comment]
>That's designed in purpose so the service worker cache doesn't mess with your hot module reloading and stuff.
Service workers do more than just caching.  In order to test background sync and push notifications, is there a way to avoid abandoning hot reloading for manual prod rebuilding?

**Comment by ndarilek** [comment]
Did anyone ever find a solution for using service workers in development mode? That is, getting all the dev mode advantages plus a service worker? I'm in the exact same boat--using them in part for push notifications, and finding it hard to test my app because I have to rebuild in production to test those features. I've already tweaked Workbox's caching to check the network first, so in theory I've mitigated the issues caused by hot reloading.

Thanks.
